astropy/table/table.py

```diff
@@ -2021,6 +2021,70 @@ def group_by(self, keys):
 
         return groups.table_group_by(self, keys)
 
+    def to_pandas(self):
+        """
+        Return a :class:`pandas.DataFrame` instance
+
+        Returns
+        -------
+        dataframe : :class:`pandas.DataFrame`
+            A pandas :class:`pandas.DataFrame` instance
+        """
+        from pandas import DataFrame
+
+        if self.has_mixin_columns:
+            raise ValueError("Cannot convert a table with mixin columns to a pandas DataFrame")
+
+        for column in self.columns:
```

Comment:
The issue is that the pandas error messages are not always very explicit. For the mixin columns it might actually work but just be very inefficient. So I prefer to control it at this stage. We *could* put it in a ``try...except`` but I'm not sure if it's easy to know what the issue is so as to know what better error to raise. I can look into it, but would be interested in @taldcroft's opinion here too.